pvlib/tests/iotools/test_psm3.py

```diff
@@ -50,47 +65,47 @@ def assert_psm3_equal(header, data, expected):
+                                 PVLIB_EMAIL, names='tmy-2017')
     expected = pd.read_csv(TMY_TEST_DATA)
     assert_psm3_equal(header, data, expected)
-    # check errors
-    with pytest.raises(HTTPError):
-        # HTTP 403 forbidden because api_key is rejected
-        psm3.get_psm3(LATITUDE, LONGITUDE, api_key='BAD', email=PVLIB_EMAIL)
-    with pytest.raises(HTTPError):
-        # coordinates were not found in the NSRDB
-        psm3.get_psm3(51, -5, DEMO_KEY, PVLIB_EMAIL)
-    with pytest.raises(HTTPError):
-        # names is not one of the available options
-        psm3.get_psm3(LATITUDE, LONGITUDE, DEMO_KEY, PVLIB_EMAIL, names='bad')
 
 
 @needs_pandas_0_22
-def test_get_psm3_singleyear():
+@pytest.mark.flaky(reruns=5, reruns_delay=2)
+def test_get_psm3_singleyear(nrel_api_key):
     """test get_psm3 with a single year"""
-    header, data = psm3.get_psm3(LATITUDE, LONGITUDE, DEMO_KEY, PVLIB_EMAIL,
-                                 names='2017', interval=30)
+    header, data = psm3.get_psm3(LATITUDE, LONGITUDE, nrel_api_key,
+                                 PVLIB_EMAIL, names='2017', interval=30)
     expected = pd.read_csv(YEAR_TEST_DATA)
     assert_psm3_equal(header, data, expected)
-    # check leap day
-    _, data_2012 = psm3.get_psm3(LATITUDE, LONGITUDE, DEMO_KEY, PVLIB_EMAIL,
-                                 names='2012', interval=60, leap_day=True)
-    assert len(data_2012) == (8760+24)
-    # check errors
-    with pytest.raises(HTTPError):
-        # HTTP 403 forbidden because api_key is rejected
-        psm3.get_psm3(LATITUDE, LONGITUDE, api_key='BAD', email=PVLIB_EMAIL,
-                      names='2017')
-    with pytest.raises(HTTPError):
-        # coordinates were not found in the NSRDB
-        psm3.get_psm3(51, -5, DEMO_KEY, PVLIB_EMAIL, names='2017')
-    with pytest.raises(HTTPError):
-        # intervals can only be 30 or 60 minutes
-        psm3.get_psm3(LATITUDE, LONGITUDE, DEMO_KEY, PVLIB_EMAIL, names='2017',
-                      interval=15)
+
+
+@pytest.mark.parametrize('latitude, longitude, api_key, names, interval',
+                         [(LATITUDE, LONGITUDE, 'BAD', 'tmy-2017', 60),
+                          (51, -5, nrel_api_key, 'tmy-2017', 60),
+                          (LATITUDE, LONGITUDE, nrel_api_key, 'bad', 60),
+                          (LATITUDE, LONGITUDE, nrel_api_key, '2017', 15),
+                          ])
+@needs_pandas_0_22
+@pytest.mark.flaky(reruns=5, reruns_delay=2)
+def test_get_psm3_tmy_errors(
+    latitude, longitude, api_key, names, interval
+):
+    """Test get_psm3() for multiple error scenarios:
+        * Bad api key -> HTTP 403 forbidden because api_key is rejected
+        * Bad latitude/longitude -> Coordinates were not found in the NSRDB
+        * Bad name -> names is not one of the available options
+        * Bad interval, single year -> intervals can only be 30 or 60 minutes
+    """
+    with pytest.raises(HTTPError) as e:
```

Comment:
NIT: please use verbose names, instead of single letters for variable names. I'm okay with either `exc` or `err` or anything longer, just not `e` -- thanks!

Reply:
I did a `grep "as e:" pvlib/**/*.py"` and found that there were two instances of that not including this one. I repeated for exc which had none and err which also had two. I decided the tie breaker would be [the pytest docs that describe](https://docs.pytest.org/en/latest/assert.html)  `pytest.raises()` that used excinfo.